maven-core/src/test/java/org/apache/maven/lifecycle/internal/stub/MojoExecutorStub.java

```diff
@@ -47,24 +48,23 @@ public MojoExecutorStub(
             BuildPluginManager pluginManager,
             MavenPluginManager mavenPluginManager,
             LifecycleDependencyResolver lifeCycleDependencyResolver,
-            ExecutionEventCatapult eventCatapult )
+            ExecutionEventCatapult eventCatapult,
+            MojosExecutionStrategy mojosExecutionStrategy )
     {
-        super( pluginManager, mavenPluginManager, lifeCycleDependencyResolver, eventCatapult );
+        super( pluginManager, mavenPluginManager, lifeCycleDependencyResolver, eventCatapult, mojosExecutionStrategy );
     }
 
     @Override
-    public void execute( MavenSession session, MojoExecution mojoExecution, ProjectIndex projectIndex,
-                         DependencyContext dependencyContext, PhaseRecorder phaseRecorder )
+    public void execute( MavenSession session, List<MojoExecution> mojoExecutions, ProjectIndex projectIndex )
         throws LifecycleExecutionException
     {
-        executions.add( mojoExecution );
+        executions.addAll( mojoExecutions );
     }
 
     @Override
-    public void execute( MavenSession session, List<MojoExecution> mojoExecutions, ProjectIndex projectIndex )
-        throws LifecycleExecutionException
+    public List<MavenProject> executeForkedExecutions( MojoExecution mojoExecution, MavenSession session, ProjectIndex projectIndex ) throws LifecycleExecutionException
     {
-        executions.addAll(mojoExecutions);
+        return null;
```

Comment:
Can you please explain those two changes. Why is the first one now `addAll` and the second one `null`?

Reply:
The `MojoExecutor` class is modified by the PR, so we need to adjust the corresponding stub.
The two methods that are overriden are actually not the same than before, as you can see, both signatures have changed.  The first method became private and the stub now overrides a higher level one, which has a list as a parameter instead of a single `MojoExecution`.  The second method is unrelated and not returns null as a stub.